synfig-studio/src/gui/cellrenderer/cellrenderer_value.cpp

```diff
@@ -172,6 +173,28 @@ class studio::ValueBase_Entry : public Gtk::CellEditable, public Gtk::EventBox
 		SYNFIG_EXCEPTION_GUARD_END_BOOL(true)
 	}
 
+	bool on_key_press_event(GdkEventKey* key_event)
+	{
+		SYNFIG_EXCEPTION_GUARD_BEGIN()
+		if(key_event->keyval == GDK_KEY_Escape)
+		{
+			hide();
+			if (parent) parent->grab_focus();
+			if(!property_editing_canceled())
```

Comment:
I think better to move this condition back to `CellRenderer_ValueBase::on_value_editing_done`.
In this case, everything will be very simple: 
```
bool on_key_press_event(GdkEventKey* key_event)
{
  SYNFIG_EXCEPTION_GUARD_BEGIN()
  if(key_event->keyval == GDK_KEY_Escape)
  {
	  hide();
	  if (parent) parent->grab_focus();
	  property_editing_canceled() = true;
	  Gtk::CellEditable::editing_done();
	  return true;
  }
  return Gtk::EventBox::on_key_press_event(key_event);
  SYNFIG_EXCEPTION_GUARD_END_BOOL(true)
}

void CellRenderer_ValueBase::on_value_editing_done() 
{
	if (value_entry && value_entry->property_editing_canceled()) {
		return;
	}
	...
```

Also, there is no need for a second check of `property_editing_canceled()`, because in that case you are already exited.